dashboard/app/models/user.rb

```diff
@@ -1233,6 +1273,16 @@ def script_level_hidden?(script_level)
     end
   end
 
+  def visible_script_levels(script)
+    visible_sls = []
+    script.script_levels.each do |sl|
```

Comment:
note this could more simply be written as:

```ruby
def visible_script_levels(script)
  script.script_levels.select do |sl|
    !script_level_hidden?(sl)
  end
end